python/cudf_polars/cudf_polars/containers/dataframe.py

```diff
@@ -0,0 +1,173 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+
+"""A dataframe, with some properties."""
+
+from __future__ import annotations
+
+from functools import cached_property
+from typing import TYPE_CHECKING
+
+import polars as pl
+
+import cudf._lib.pylibcudf as plc
+
+from cudf_polars.containers.column import Column
+
+if TYPE_CHECKING:
+    from collections.abc import Mapping, Sequence, Set
+
+    from typing_extensions import Self
+
+    import cudf
```

Comment:
Let's mark all the places we're importing/using cudf rather than pylibcudf with TODOs or something so that we can find them easily as we work to remove them.